packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,2896 @@
+/// * [MenuItemGroup], a menu item which collects its members into a group
+///   separated from other menu items by a divider.
+/// * [MenuItemButton], a leaf menu item which displays the label, an optional
+///   shortcut label, and optional leading and trailing icons.
+/// * [createCascadingMenu], a function that creates a [MenuEntry] that allows
+///   creation and management of a cascading menu anywhere.
+/// * [MenuController], a class that allows controlling and connecting menus.
+/// * [PlatformMenuBar], which creates a menu bar that is rendered by the host
+///   platform instead of by Flutter (on macOS, for example).
+/// * [ShortcutRegistry], a registry of shortcuts that apply for the entire
+///   application.
+/// * [VoidCallbackIntent] to define intents that will call a [VoidCallback] and
+///   work with the [Actions] and [Shortcuts] system.
+/// * [CallbackShortcuts] to define shortcuts that simply call a callback and
+///   don't involve using [Actions].
+class MenuBar extends StatefulWidget with DiagnosticableTreeMixin {
+  /// Creates a const [MenuBar].
+  const MenuBar({
+    super.key,
+    this.controller,
+    this.backgroundColor,
+    this.minimumHeight,
+    this.padding,
+    this.elevation,
+    this.shape,
+    this.expand = true,
+    this.children = const <Widget>[],
+  });
+
+  /// An optional controller that allows outside control of the menu bar.
+  ///
+  /// You can use a controller to close any open menus from outside of the menu
+  /// bar using [MenuController.closeAll].
+  final MenuController? controller;
+
+  /// The background color of the menu bar.
+  ///
+  /// This is a [MaterialStateProperty], but [MenuBar] doesn't currently have
+  /// any states for it to respond to. Use [MaterialStatePropertyAll] to
+  /// initialize it.
+  ///
+  /// Defaults to [MenuThemeData.barBackgroundColor] if null.
+  final MaterialStateProperty<Color?>? backgroundColor;
+
+  /// The preferred minimum height of the menu bar.
+  ///
+  /// Defaults to the value of [MenuThemeData.barMinimumHeight] if null.
+  final double? minimumHeight;
+
+  /// The padding around the contents of the menu bar itself.
+  ///
+  /// Defaults to the value of [MenuThemeData.barPadding] if null.
+  final EdgeInsetsDirectional? padding;
+
+  /// The shape of the [MenuBar]'s border.
+  ///
+  /// This is a [MaterialStateProperty], but [MenuBar] doesn't currently have
+  /// any states for it to respond to. Use [MaterialStatePropertyAll] to
+  /// initialize it.
+  ///
+  /// Default to [MenuThemeData.barShape] if null.
+  final MaterialStateProperty<ShapeBorder?>? shape;
+
+  /// If true, the menu bar expands horizontally to fill available space.
+  ///
+  /// Defaults to true.
+  final bool expand;
+
+  /// The Material elevation of the menu bar (if any).
+  ///
+  /// This is a [MaterialStateProperty], but [MenuBar] doesn't currently have
+  /// any states for it to respond to. Use [MaterialStatePropertyAll] to
```

Comment:
You could choose to say that the menu bar is focused when any menu bar item is focused. Likewise selected when any menu bar menu is open.